packages/grid/_modules_/grid/hooks/features/useColumnResize.tsx

```diff
@@ -1,204 +1,130 @@
-      updateWidth(currentColDefRef.current.width! + 10);
-      mouseLeftTimeout.current = setTimeout(() => {
-        stopResize();
-      }, MOUSE_LEFT_TIMEOUT);
-    } else if (isResizing) {
-      mouseLeftTimeout.current = setTimeout(() => {
-        stopResize();
-      }, MOUSE_LEFT_TIMEOUT);
+    });
+
+    logger.debug(
+      `Updating col ${colDefRef.current!.field} with new width: ${colDefRef.current!.width}`,
+    );
+  });
+
+  const handleResizeMouseMove = useEventCallback((nativeEvent) => {
+    // Cancel move in case some other element consumed a mouseup event and it was not fired.
+    if (nativeEvent.buttons === 0) {
+      // @ts-expect-error fixed in v5
+      handleResizeMouseUp(nativeEvent);
+      return;
     }
-  }, [headerHeight, logger, stopResize, updateWidth]);
-
-  const handleMouseMove = React.useCallback(
-    (ev: MouseEvent): void => {
-      if (isResizing.current) {
-        const target = ev.currentTarget! as HTMLDivElement;
-        const rect = target.getBoundingClientRect();
 
-        resizingMouseMove.current = { x: ev.clientX - rect.left, y: ev.clientY - rect.top };
-
-        const offsetLeft = !isLastColumn.current ? rect.left : scrollOffset.current * -1;
-
-        let newWidth = ev.clientX - offsetLeft - currentColPosition.current!;
-        newWidth = newWidth > MIN_COL_WIDTH ? newWidth : MIN_COL_WIDTH;
-        updateWidth(newWidth);
-      }
-    },
-    [updateWidth],
-  );
-
-  // This a hack due to the limitation of react as I cannot put columnsRef in the dependency array of the effect adding the Event listener
-  const columnsRefState = useStateRef(columnsRef);
+    let newWidth =
+      initialOffset.current +
+      nativeEvent.clientX -
+      colElementRef.current!.getBoundingClientRect().left;
+    newWidth = Math.max(MIN_COL_WIDTH, newWidth);
+
+    updateWidth(newWidth);
+  });
+  const handleMouseDown = useEventCallback((event: React.MouseEvent<HTMLDivElement>) => {
+    // Only handle left clicks
+    if (event.button !== 0) {
+      return;
+    }
 
-  React.useEffect(() => {
-    if (columnsRef && columnsRef.current) {
-      logger.info('Adding resizing event listener');
-      const columnsRefEvents = columnsRef.current;
-      columnsRef.current.addEventListener('mouseup', stopResize);
-      columnsRef.current.addEventListener('mouseleave', handleMouseLeave);
-      columnsRef.current.addEventListener('mouseenter', handleMouseEnter);
-      columnsRef.current.addEventListener('mousemove', handleMouseMove);
-
-      return () => {
-        columnsRefEvents.removeEventListener('mouseup', stopResize);
-        columnsRefEvents.removeEventListener('mouseleave', handleMouseLeave);
-        columnsRefEvents.removeEventListener('mouseenter', handleMouseEnter);
-        columnsRefEvents.removeEventListener('mousemove', handleMouseMove);
-      };
+    if (!event.currentTarget.classList.contains('MuiDataGrid-columnSeparatorResizable')) {
```

Comment:
Can we combine this `if` statement with the one before it as both of them if `true` exit the function?

Reply:
How about we add a comment to bring clarity?

```suggestion
    // Skip if the column isn't resizable
    if (!event.currentTarget.classList.contains('MuiDataGrid-columnSeparatorResizable')) {
```